### ISSUE TITLE
[REQUEST]Automatic switch theme from dark to light based on Gnome theme

### ISSUE THREAD

**Issue by lamyergeier** [aristocratos/btop#928]
**Is your feature request related to a problem? Please describe.**
I would like to switch from solarized light to solarized dark depending on if on Gnome it is dark theme enabled or not. 

**Describe the solution you'd like**
Automatic switch theme from dark to light based on Gnome theme, or atleast when it is starting it should be mindful of the Gnome dark theme settings

# Additional

Please give CLI option to set theme, so that I can atleast automate through a bash function

**Comment by imwints** [comment]
The first problem is that there is no mapping of [Light <-> Dark] themes. There are a couple of themes that only have one variant. So btop doesn't know (and probably shouldn't care) if you're theme is dark or light. The other thing is that this would tie a CLI application to a desktop, and I'm not sure if this is good idea.

> Please give CLI option to set theme, so that I can atleast automate through a bash function

This seems reasonable.

As a workaround, this wrapper script edits the config based on the theme type you provide it.
You still need to talk to gnome and parse it's mode and pass either `dark` or `light` to the script.

```python
#!/usr/bin/env python3

import os
import subprocess
import sys
from pathlib import Path

# Mode is the first command line argument
# As an alternative, set mode based on gnome's theme type here
mode = sys.argv[1]

# Path to btop's config file in the user's home
config = (
    Path(os.path.expanduser("~"))
    .joinpath(".config")
    .joinpath("btop")
    .joinpath("btop.conf")
)

# Read the config, replacing the `color_theme` statement based on the `mode`
with config.open() as file:
    lines = [
        line if "color_theme" not in line else f'color_theme = "solarized_{mode}"\n'
        for line in file
    ]

# Write the new config
with config.open("w") as file:
    for line in lines:
        file.write(line)

# Launch btop
subprocess.run(["btop"], check=False)
```

**Comment by lamyergeier** [comment]
@imwints Could you provide a CLI Option --theme?

So users can start btop as follows

btop --theme "Solarized dark"

**Comment by TheurgicDuke771** [comment]
+1 for the [workaround](https://github.com/aristocratos/btop/issues/928#issuecomment-2361039253). 
Instead of passing the `mode` value manually every time you can use a third party library [darkdetect](https://github.com/albertosottile/darkdetect), and replace this below code: 
```
mode = sys.argv[1]
``` 
with this if block: 
```
if darkdetect.isDark():
    mode = 'dark'
else:
    mode = 'light'
```

**Comment by alexis-opolka** [comment]
I agree with @lamyergeier on an argument to be able to override the configuration via the command line with `--theme`.
Would it be possible @deckstose ?

I can try to make a PR for that if you'd like.

**Comment by HosseyNJF** [comment]
Hello! @deckstose , if you give the green light, I'm ready to contribute the `--theme` CLI flag. I think it's a nice feature to have, you'd essentially just have to define an alias like `btop --theme=$([ <whatever condition>  ] && echo "dark" || echo "light")` instead of the wrapper script.

**Comment by comf4rtnoise** [comment]
For Mac OS only:

I use [https://github.com/cormacrelf/dark-notify](url) to detect system theme mode. 

File tree of my config_dir:
```bash
.
├── btop.conf
├── btop_dark.conf
├── btop_light.conf
└── themes
    ├── catppuccin_frappe.theme
    ├── catppuccin_latte.theme
    ├── catppuccin_macchiato.theme
    └── catppuccin_mocha.theme
```

Command alias:
```bash
# btop
alias btop='btop -c ${HOME}/.config/btop/btop_$(dark-notify -e).conf'
```

**Comment by mcnesium** [comment]
If the goal of OP is only to have no dark background on light system theme or vice versa, then just set `theme_background = False` in the config and btop will just use whatever colors the terminal uses. This also works great if btop is run in a remote shell.
